backend/capellacollab/projects/toolmodels/modelsources/git/github/interface.py

```diff
@@ -0,0 +1,197 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+import io
+import json
+import shutil
+import tempfile
+import typing as t
+import zipfile
+from urllib import parse
+
+import fastapi
+import requests
+from fastapi import status
+
+import capellacollab.projects.toolmodels.modelsources.git.models as git_models
+import capellacollab.settings.modelsources.git.models as settings_git_models
+
+
+def get_project_id_by_git_url(
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> str:
+    # Project ID contains /{owner}/{repo_name}
+    del git_instance  # unused
```

Comment:
Shall I then write something that the interface should be the same? otherwise it might be at first sight a bit weird that there is a variable which first thing in the method is deleted?